### DISCUSSION TITLE
YouTube Links Still Broken

### DISCUSSION THREAD

**Discussion by alima566** [Androz2091/discord-player#687] [Q&A]
**Describe the bug**
Copying and pasting a YouTube link gives me "I could not find track" error message. Bot joins the VC, but doesn't play anything.

**Screenshots**
![image](https://user-images.githubusercontent.com/14950662/128730903-213f59ba-50bb-463d-b2b8-72b7f49750e2.png)

**Please complete the following information:**
 - Node Version: 16.6.1
 - Discord Player Version: 5.0.1
 - Discord.js Version: 13.0.1

**Comment by twlite** [comment]
Hello @alima566, I am unable to reproduce your issue

![image](https://user-images.githubusercontent.com/46562212/128732145-407d217e-0684-44b9-ba0c-c71cdbea0a2f.png)

I did this test in example bot https://github.com/Androz2091/discord-player/tree/master/example/music-bot

**Comment by DawnOfDedSec** [comment]
Yeah Same Issue Facing by me too , code was not changed and Updated to discord-player@dev and updated every package to latest version . But then these Issue came in . I tested on example bot , and it was working , but not on my bot , But code was Same more or less

**Comment by DawnOfDedSec** [comment]
```
2021-08-14-07:33:42 0|index  | TypeError: soundcloud_scraper_1.Client.getPlaylist is not a function
2021-08-14-07:33:42 0|index  |     at Player.<anonymous> (/home/container/node_modules/discord-player/dist/Player.js:380:68)
2021-08-14-07:33:42 0|index  |     at Generator.next (<anonymous>)
2021-08-14-07:33:42 0|index  |     at fulfilled (/home/container/node_modules/discord-player/dist/Player.js:5:58)
2021-08-14-07:33:42 0|index  |     at runMicrotasks (<anonymous>)
2021-08-14-07:33:42 0|index  |     at processTicksAndRejections (node:internal/process/task_queues:94:5)
```

Error Coming in Soundcloud link

**Comment by umutdeveci** [comment]
Hey - I think I found a way to reproduce this bug: If the searched URL points to an age restricted URL, the search results do not return anything from Youtube. The reasons seems to be that `ytdlOptions` are not passed through to the search option, regardless of `cookie` parameter present or not.

https://github.com/Androz2091/discord-player/blob/94ffd93cde2257034122c98e03ffdb9582e15233/src/Player.ts#L227

A small patch should fix this, by just passing the `ytdlOptions` for the search call, when the video is age restricted and the bot has access to the youtube cookie.
